subgraphs/aave-forks/protocols/spark-lend/src/mapping.ts

```diff
@@ -0,0 +1,659 @@
+
+export function updateMarketRewards(
+  event: ethereum.Event,
+  market: Market,
+  rewardToken: RewardToken,
+  emissionRate: BigInt
+): void {
+  let rewardTokens = market.rewardTokens;
+  let rewardTokenEmissions = market.rewardTokenEmissionsAmount;
+  let rewardTokenEmissionsUSD = market.rewardTokenEmissionsUSD;
+
+  if (rewardTokens == null) {
+    rewardTokens = [];
+    rewardTokenEmissions = [];
+    rewardTokenEmissionsUSD = [];
+  }
+
+  let rewardTokenIndex = rewardTokens.indexOf(rewardToken.id);
+  if (rewardTokenIndex == -1) {
+    rewardTokenIndex = rewardTokens.push(rewardToken.id) - 1;
+    rewardTokenEmissions!.push(BIGINT_ZERO);
+    rewardTokenEmissionsUSD!.push(BIGDECIMAL_ZERO);
+  }
+  rewardTokenEmissions![rewardTokenIndex] = emissionRate.times(
+    BigInt.fromI32(SECONDS_PER_DAY)
+  );
+  market.rewardTokens = rewardTokens;
+  market.rewardTokenEmissionsAmount = rewardTokenEmissions;
+  market.rewardTokenEmissionsUSD = rewardTokenEmissionsUSD;
+  market.save();
+
+  sortRewardTokens(market);
+  updateMarketRewardEmissions(market, event);
+}
+
+function updateMarketRewardEmissions(
+  market: Market,
+  event: ethereum.Event
+): void {
+  if (market.rewardTokens == null) {
+    return;
+  }
+
+  const protocol = getOrCreateLendingProtocol(protocolData);
+  const rewardTokens = market.rewardTokens!;
+  const rewardEmissions = market.rewardTokenEmissionsAmount!;
+  const rewardEmissionsUSD = market.rewardTokenEmissionsUSD!;
+  for (let i = 0; i < rewardTokens.length; i++) {
+    const rewardToken = RewardToken.load(rewardTokens[i])!;
+    if (event.block.timestamp.gt(rewardToken._distributionEnd!)) {
+      rewardEmissions[i] = BIGINT_ZERO;
+      rewardEmissionsUSD[i] = BIGDECIMAL_ZERO;
+    } else {
+      const token = Token.load(rewardToken.token)!;
+      let rewardTokenPriceUSD = token.lastPriceUSD;
+      rewardTokenPriceUSD = getAssetPriceInUSDC(
+        Address.fromString(token.id),
+        Address.fromString(protocol._priceOracle)
+      );
+
+      rewardEmissionsUSD[i] = rewardEmissions[i]
+        .toBigDecimal()
+        .div(exponentToBigDecimal(token.decimals))
+        .times(rewardTokenPriceUSD);
+    }
+  }
+  market.rewardTokenEmissionsAmount = rewardEmissions;
+  market.rewardTokenEmissionsUSD = rewardEmissionsUSD;
+  market.save();
+}
+
+function sortRewardTokens(market: Market): void {
+  if (market.rewardTokens!.length <= 1) {
+    return;
+  }
+
+  const tokens = market.rewardTokens;
+  const emissions = market.rewardTokenEmissionsAmount;
+  const emissionsUSD = market.rewardTokenEmissionsUSD;
+  multiArraySort(tokens!, emissions!, emissionsUSD!);
+
+  market.rewardTokens = tokens;
+  market.rewardTokenEmissionsAmount = emissions;
+  market.rewardTokenEmissionsUSD = emissionsUSD;
+  market.save();
+}
+
+function multiArraySort(
+  ref: Array<string>,
+  arr1: Array<BigInt>,
+  arr2: Array<BigDecimal>
+): void {
+  if (ref.length != arr1.length || ref.length != arr2.length) {
+    // cannot sort
```

Comment:
maybe log an error instead of return silently?

Reply:
good call